API-Samples/Java/app/src/main/java/videoindexersamples/VideoIndexerClient.java

```diff
@@ -0,0 +1,239 @@
+                    .POST(HttpRequest.BodyPublishers.ofString(accessTokenRequestStr))
+                    .build();
+
+            var response = httpStringResponse(request);
+            AccessTokenResponse accessTokenResponse = gson.fromJson(response.body(), AccessTokenResponse.class);
+            this.accountAccessToken = accessTokenResponse.accessToken;
+        } catch (URISyntaxException | IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Uploads a video and starts the video index. Calls the uploadVideo API (<a href="https://api-portal.videoindexer.ai/api-details#api=Operations&operation=Upload-Video">...</a>)
+     *
+     * @param videoUrl : the video Url to upload
+     * @return Video Id of the video being indexed, otherwise throws exception
+     */
+    public String uploadVideo(String videoUrl, String videoName) {
+
+        Map<String, String> map = new HashMap<>();
+        map.put("accessToken", this.accountAccessToken);
+        map.put("name", videoName);
+        map.put("description", "video_description");
+        map.put("privacy", "private");
+        map.put("partition", "partition");
+        map.put("videoUrl", videoUrl);
+        var queryParam = Utils.toQueryParamString(map);
+
+        var requestUri = String.format("%s/%s/Accounts/%s/Videos?%s", ApiUrl, account.location, account.properties.accountId, queryParam);
+
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI(requestUri))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.noBody())
+                    .build();
+
+            var response = httpStringResponse(request);
+            Video upoloadedVideo = gson.fromJson(response.body(), Video.class);
+
+            String videoId = upoloadedVideo.id;
+            System.out.printf("Video ID %s was uploaded successfully.\n", videoId);
+            return videoId;
+        } catch (URISyntaxException | IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    /**
+     * Searches for the video in the account. Calls the searchVideo API (<a href="https://api-portal.videoindexer.ai/api-details#api=Operations&operation=Search-Videos">...</a>)
+     *
+     * @param videoId The video id
+     * @return Video Metadata
+     */
+    public String getVideo(String videoId) {
+        System.out.printf("Searching videos in account %s for video ID %s.\n", account.properties.accountId, videoId);
+
+        Map<String, String> map = new HashMap<>();
+        map.put("accessToken", this.accountAccessToken);
+        map.put("id", videoId);
+        var queryParam = Utils.toQueryParamString(map);
+
+        try {
+            var requestUri = String.format("%s/%s/Accounts/%s/Videos/Search?%s", ApiUrl, account.location, account.properties.accountId, queryParam);

```

Comment:
Why use search to get the video? There is a GetVideo API

Reply:
this was the sample they did on C#
there is a GetVideo sample as well